### DISCUSSION TITLE
Use cases that might need archive node?

### DISCUSSION THREAD

**Discussion by uniyj** [ponder-sh/ponder#156] [General]
Hi @0xOlias. It was great chatting with you yesterday. Thought I'll just post here the following query that we briefly discussed, but couldn't conclude (because I don't fully know what I am talking about, most of the time =P )

Are there use cases for ponder that might require an archive node? Or perhaps cases where it can benefit (eg in speed/performance) from having an archive node vs a normal full node?

Curious to hear others' thoughts on these. Super cool to see new things coming up in the space. Thanks!

yj

**Comment by typedarray** [comment]
My understanding is that the only difference between a full node and an archive node is that an archive node is able to efficiently query contract state and account balances at _any_ block height, while a full node must reconstruct that point-in-time state from historical block data (which can be computationally expensive). The RPC methods in question here are any that accept a "time-travel" `blockTag` parameter: `eth_call`, `eth_getBalance`, `eth_getCode`, etc. At the moment, Ponder's core does not use any of these methods, so it works with a full node.

However, some Ponder apps do make historical `eth_call` requests to read [historical contract state](https://ponder.sh/api-reference/event-handlers#readonlycontract), which will require an archive node.

Ideally, most Ponder apps don't need to read historical contract state because all indexed contracts emit events including all the data necessary to build the desired API. In practice, this is difficult to achieve.
